### PR TITLE
bauhaus: fix background & borders 

### PR DESCRIPTION
**Pull request by TurboGit**

### CODE REVIEW

**Comment by TurboGit** [comment]
@Nilvus : Please can you review this one ? I have found how to remove the black background and have a small border for the combo.

**Comment by Nilvus** [comment]
I will test ASAP but that seems more a workaround as it should not be necessary to add a new CSS setting. That's the problem I see on your code. All background and borders are already set by my last PR, even for those bauhaus widget. The only thing is to see how that could be correctly applied. New CSS setting is not good here.

**Comment by Nilvus** [comment]
I see that you draw background and border with cairo. The better way would be to not draw that with cairo but with CSS. But maybe it's not possible for those widget?

**Comment by TurboGit** [comment]
No it's not possible as we are using a Cairo surface where everything is drawn. Yet the border color is using a CSS definition. I.e. The colour is not hard coded.

**Comment by TurboGit** [comment]
@Nilvus : An alternative to borders in popup if you don't like them is to have all popup window background really different for the window to standout where it opens.

File: data/themes/darktable.css [hunk 1]
```diff
@@ -855,6 +855,12 @@ dialog .dialog-vbox notebook stack box box checkbutton label  /* fix a margin on
   color: @bauhaus_fg;
 }
 
+#bauhaus-combobox *,
+#bauhaus-slider *
```
Comment (Nilvus) [inline data/themes/darktable.css:859]: No need to set a specific background here. To be sure background it's the same as other popover menus, add those 2 bauhaus lines around line 900. Around line 900, that will give you that:

```
context-menu,
menu,
menuitem > menu, /* sub-menu */
menuitem arrow,
tooltip,
popover,
combobox window,
dialog combobox window,
#bauhaus-combobox *,
#bauhaus-slider *,
#modulegroups-iop-popup
{
  opacity: 1;  /* this is needed for tooltip on/off feature with Shift+t shortcut */
  background-color: @tooltip_bg_color;
  border-radius: 2px;
  border: 1px solid @button_border;
  outline-style: none;
  min-height: 0.4em;
  min-width: 0.4em;
  padding: 0.2em 0;
}
```
Settings allow to have consistency for all that items. Tested on my side and all good

**Review by Nilvus — COMMENTED** [review]
Other things to improve are 2 glitch remaining:

- for example, a right-click on exposure slider. See the border missing on left of the slider and the missing space between border and exposure name



- on popover bauhaus-combobox, I have a  double border at the bottom. I have difficulties to find how to remove it. It's seems that a border is set by CSS as with CSS change commited here, if I set border to red color, I see that the second bottom border is red. But if I remove that CSS border setting, this second bottom border became black. So I think it's something remaining in Gtk code here.

**Comment by Nilvus** [comment]
Oh another thing: did you see that this PR is not on a specific branch but on your master?

**Comment by TurboGit** [comment]
>     * See the border missing on left of the slider and the missing space between border and exposure name

Yes, because the border is a rectangle around the window, but it seems that it is written over by the widget.

> Oh another thing: did you see that this PR is not on a specific branch but on your master?

Yes, this is done directly on my master and I push it on my clone repo. I should just pay attention to not push on origin :)

**Comment by TurboGit** [comment]
@Nilvus : thanks for the help, a nightmare those borders :( but I think I got this right now. Can you double check ? I think this is ready to be merged at this time, just waiting your go.

**Comment by Nilvus** [comment]
Oh yes, those bauhaus widget are a nightmare. But, good news, we got it. All good. Good work. Bauhaus widgets was the last thing we couldn't tweak before than now are. Great to see, we polish again more the UI. Thanks!

**Comment by TurboGit** [comment]
Thanks for review!

**Pull request merged**